pkg/providers/gcore/client.go

```diff
@@ -0,0 +1,86 @@
+package gcore
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+
+	gdns "github.com/G-Core/gcore-dns-sdk-go"
+	phonebook "github.com/pier-oliviert/phonebook/api/v1alpha1"
+)
+
+const (
+	EnvAPIURL   = "GCORE_API_URL"
+	EnvAPIToken = "GCORE_API_TOKEN"
+	DefaultTTL  = int64(120) // gcore doesn't support shorter TTL for the free plan, so 120 is the basis to avoid confusions
+)
+
+// This interface is created so mocks can be done on testing. Since gcore doesn't have any interface to work with,
+// this needs to exists here in order for phonebook to have proper testing
+type api interface {
+	AddZoneRRSet(context.Context, string, string, string, []gdns.ResourceRecord, int, ...gdns.AddZoneOpt) error
+	DeleteRRSet(context.Context, string, string, string) error
+}
+
+type gcore struct {
+	integration string
+	zoneID      string
+	zones       []string
+	api         api
+}
+
+func NewClient(ctx context.Context) (*gcore, error) {
+	var err error
+
+	apiURL := os.Getenv(EnvAPIURL)
+
+	token := os.Getenv(EnvAPIToken)
```

Comment:
Should these be set from the normal method? Apologies if I've missed it (on my phone ATM and GitHub isn't the greatest 😂) but I've been using the `utils.RetrieveValueFromEnvOrFile` function to pull these in

Reply:
Honestly, I copied these from https://github.com/G-Core/external-dns-gcore-webhook/blob/master/gcoreprovider/gcore.go with no second thoughts. I'll make the changes to use `utils.RetrieveValueFromEnvOrFile`.